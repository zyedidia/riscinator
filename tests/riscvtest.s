main: addi x2, x0, 5
addi x3, x0, 12
addi x7, x3, -9
or x4, x7, x2
and x5, x3, x4
add x5, x5, x4
beq x5, x7, end
slt x4, x3, x4
beq x4, x0, around
addi x5, x0, 0
around: slt x4, x7, x2
add x7, x4, x5
sub x7, x7, x2
sw x7, 84(x3)
lw x2, 96(x0)
add x9, x2, x5
jal x3, end
addi x2, x0, 1
end: add x2, x2, x9
sw x2, 0x20(x3)
done: j done
