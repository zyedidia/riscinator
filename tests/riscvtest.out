x0: 0
x1: 0
x2: 25
x3: 68
x4: 1
x5: 11
x6: 0
x7: 7
x8: 0
x9: 18
x10: 0
x11: 0
x12: 0
x13: 0
x14: 0
x15: 0
x16: 0
x17: 0
x18: 0
x19: 0
x20: 0
x21: 0
x22: 0
x23: 0
x24: 0
x25: 0
x26: 0
x27: 0
x28: 0
x29: 0
x30: 0
x31: 0
